src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -3029,14 +3041,17 @@ int32_t NaClSysShmat(struct NaClAppThread  *natp,
   }
 
   if (length > 0) {
+    //we hack the file_size field to store the shmid, because FILE_DESC_MAX is less than 4096 
```

Comment:
So Id much rather add an entry struct field than to hack the length here. Even if its neat and works, seems like it will be really confusing down the road.